### DISCUSSION TITLE
How to use persist plugin inside the store?

### DISCUSSION THREAD

**Discussion by vladninja** [alpinejs/alpine#2176] [1. Help] [answered]
Hi. I want to use persist plugin inside the store.

I created a store:
`export default {
    works: [],
    activeSlides: [],
    activeImageIndex: null,
    activeWorkIndex: swiperCarousel.activeIndex,
    favorites: this.$persist([])
    }`

but I am getting an error: "Uncaught TypeError: Cannot read properties of undefined (reading '$persist')".

Of course, I am imported and use a persist plugin:

`import Alpine from "alpinejs";
import persist from "@alpinejs/persist";
Alpine.plugin(persist);`

When I am using persist outside of store, it works well.
Could you help me?

**Comment by SimoTod** [comment, marked as answer]
https://github.com/alpinejs/alpine/discussions/2032
